secretsmanager/src/secrets_manager_impl.cpp

```diff
@@ -0,0 +1,41 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+//
+// This convenience header combines different block implementations.
+
+#include "secrets_manager_impl.h"
+
+#include <string>
+#include <fstream>
+
+namespace concord::secretsmanager {
+
+std::string ISecretsManagerImpl::readFile(std::string_view path) {
+  std::string data;
+
+  std::string buf(4096, '0');
+  std::ifstream in(path.data());
+  while (in.read(buf.data(), buf.length())) {
```

Comment:
Why are you reading into `buf` and then copying into `data` rather than just reading into `buf` repeatedly and then returning it?

Reply:
Fixed